app/routes/surveys/AddSurveyRoute.js

```diff
@@ -1,31 +1,85 @@
 import { connect } from 'react-redux';
 import { compose } from 'redux';
-import { addSurvey, deleteSurvey } from '../../actions/SurveyActions';
+import {
+  addSurvey,
+  deleteSurvey,
+  fetchTemplate
+} from '../../actions/SurveyActions';
 import SurveyEditor from './components/SurveyEditor/SurveyEditor';
 import { LoginPage } from 'app/components/LoginForm';
 import replaceUnlessLoggedIn from 'app/utils/replaceUnlessLoggedIn';
 import { push } from 'react-router-redux';
-import moment from 'moment-timezone';
 import { initialQuestion } from './components/SurveyEditor/SurveyEditor';
+import prepare from 'app/utils/prepare';
+import { selectSurveyTemplate } from 'app/reducers/surveys';
+import { fetchEvent } from 'app/actions/EventActions';
+import { selectEventById } from 'app/reducers/events';
+import { defaultActiveFrom } from './utils';
 
-const time = (hours, minutes) =>
-  moment()
-    .startOf('day')
-    .add({ hours, minutes })
-    .toISOString();
+const loadData = (props, dispatch) => {
+  const { template, event } = props.location.query;
+  if (event) {
+    return dispatch(fetchEvent(event)).then(
+      result =>
+        result.success &&
+        dispatch(fetchTemplate(result.payload.entities.events[event].eventType))
```

Comment:
if the only reason you're fetching the event is to retrieve the template, maybe we should have an API endpoint on the backend for doing so? `/api/v1/events/10/survey-template` or something like that? up to you, but might save you some code :p

Reply:
I use it for setting initialValues `survey.event` (which requires title for the autocomplete selector)  and `activeFrom = event.endTime` too. I suppose I fetch quite a bit more information about the event than I need, but I figured that wasn't a huge deal? You'll only ever do one such fetch at a time.